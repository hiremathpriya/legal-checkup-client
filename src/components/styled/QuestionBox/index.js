import styled from 'styled-components'

const QuestionBox = styled.div`
  width: 260px;
  font-family: Helvetica;
  font-size: 25px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  margin: 0px auto 10px auto;
`
QuestionBox.displayName = 'StyledQuestionBox'

export default QuestionBox
