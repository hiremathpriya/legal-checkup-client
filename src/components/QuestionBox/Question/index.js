import React from 'react'
import { StyledQuestion } from '../../styled'

import { handleFalsy } from '../../../utilities'

export default function Question ({ questions, activeQuestion }) {
  const { body } = questions[activeQuestion]

  return handleFalsy(body, <StyledQuestion>{body}</StyledQuestion>)
}
