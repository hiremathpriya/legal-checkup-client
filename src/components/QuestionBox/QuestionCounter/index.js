import React from 'react'
import { StyledQuestionCounter } from '../../styled'
import { keysIn, length } from 'ramda'

import { handleFalsy } from '../../../utilities'

export default function QuestionCounter ({
  questions = {},
  activeQuestion = null
}) {
  const questionCount = length(keysIn(questions))
  return handleFalsy(
    activeQuestion && questionCount,
    <StyledQuestionCounter>
      {`${activeQuestion} / ${questionCount}`}
    </StyledQuestionCounter>
  )
}
