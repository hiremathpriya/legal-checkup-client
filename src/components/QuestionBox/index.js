import React from 'react'

import Question from '../../containers/Question'
import QuestionCounter from '../../containers/QuestionCounter'

export default function QuestionBox () {
  return (
    <div>
      <QuestionCounter />
      <Question />
    </div>
  )
}
