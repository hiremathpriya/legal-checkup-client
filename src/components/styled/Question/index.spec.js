import React from 'react'
import { shallow } from 'enzyme'

import QuestionBox from '.'

describe('components:styled:QuestionBox', function () {
  it('renders the Layout and CSS properties properly', () =>
    expect(
      toJson(shallow(<QuestionBox>StyledQuestionBox</QuestionBox>))
    ).toMatchSnapshot())
})
