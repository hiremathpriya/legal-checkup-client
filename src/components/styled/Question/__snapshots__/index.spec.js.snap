// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:QuestionBox renders the Layout and CSS properties properly 1`] = `
.c0 {
  width: 260px;
  height: 120px;
  font-family: Helvetica;
  font-size: 25px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: center;
}

<div
  className="c0"
>
  StyledQuestionBox
</div>
`;
